src/helper/remote_config/protocol/config_state.hpp

```diff
@@ -0,0 +1,31 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+#pragma once
+
+#include <string>
+
+namespace dds::remote_config::protocol {
+
+struct config_state {
```

Comment:
As we discussed, structs are meant for simple containers. This is a container with inaccessible members, that can only be accessed through getters and constructed with a constructor. This is a class now and this comment applies to many other of the structs with a similar pattern.

Reply:
Done